internal/api/serverless_functions.go

```diff
@@ -49,16 +58,35 @@ func ServeUserApplications(okta OktaService) http.Handler {
 
 		var accounts []Application
 		for _, app := range applications {
-			if app.AppName == "amazon_aws" || strings.Contains(app.AppName, "tencent") {
-				accounts = append(accounts, Application{
-					ID:   app.AppInstanceId,
-					Name: app.Label,
-				})
+			typ, ok := deriveApplicationType(app)
+			if !ok {
+				continue
+			}
+
+			app := Application{
+				ID:   app.AppInstanceId,
+				Name: app.Label,
+				Type: typ,
+				Href: app.LinkUrl,
 			}
+
+			accounts = append(accounts, app)
 		}
 
 		requestAttrs = append(requestAttrs, slog.Int("application_count", len(accounts)))
 		slog.Info("served applications", requestAttrs...)
 		ServeJSON(w, accounts)
 	})
 }
+
+func deriveApplicationType(app *okta.AppLink) (ApplicationType, bool) {
+	if app.AppName == "amazon_aws" {
+		return ApplicationTypeAWS, true
+	}
+
+	if strings.Contains(strings.ToLower(app.AppName), "tencent") {
```

Comment:
This is likely to be changed depending on our implementation --  Okta doesn't have good first-class support for Tencent Cloud so they don't have a defined naming scheme, but we need a naming scheme to identify them.

Most likely, app.AppName will be set to something like `keyconjurer_tencent` to be able to discern keyconjurer-enabled Tencent Cloud applications from ones that just take you to the console.